File: llm_generated_code/Gemini/Python/Level2/Q8.py
def max_meetings(start_times, end_times):
    """
    Finds the maximum number of meetings that can be held and their order.

    Args:
        start_times: A list of integers representing the start times of meetings.
        end_times: A list of integers representing the end times of meetings.

    Returns:
        A list of indices of the meetings that can be held in optimal order.
    """

    n = len(start_times)
    meetings = sorted(enumerate(end_times), key=lambda x: x[1])  # Sort by end times
    
    end_time = 0
    result = []
    for i in range(n):
        if start_times[meetings[i][0]] >= end_time:
            result.append(meetings[i][0])
            end_time = end_times[meetings[i][0]]

    return result

# Example usage:
start_times = [1, 3, 0, 5, 8, 5]
end_times = [2, 4, 6, 7, 9, 9]
max_meetings = max_meetings(start_times, end_times)
print("Maximum number of meetings:", len(max_meetings))
print("Meeting order:", max_meetings)

File: llm_generated_code/Gemini/CPP/Level2/Q8.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

bool compareIntervals(const pair<int, int>& a, const pair<int, int>& b) {
    return a.second < b.second;
}

int maxMeetings(vector<int>& start, vector<int>& end) {
    int n = start.size();
    vector<pair<int, int>> meetings;
    for (int i = 0; i < n; i++) {
        meetings.push_back({start[i], end[i]});
    }

    // Sort meetings by their end times
    sort(meetings.begin(), meetings.end(), compareIntervals);

    // Time complexity: O(n log n) for sorting, O(n) for iteration
    // Space complexity: O(n) for the meetings vector
    // Lines of code: 17
    // Cyclomatic complexity: 2
    int count = 1;
    int prevEnd = meetings[0].second;
    for (int i = 1; i < n; i++) {
        if (meetings[i].first >= prevEnd) {
            count++;
            prevEnd = meetings[i].second;
        }
    }

    return count;
}

int main() {
    vector<int> start = {1, 3, 0, 5, 8, 5};
    vector<int> end = {2, 4, 6, 7, 9, 9};
    int maxMeetingsPossible = maxMeetings(start, end);
    cout << "Maximum number of meetings: " << maxMeetingsPossible << endl;

    return 0;
}
